dbms/src/Storages/KVStore/MultiRaft/Disagg/CheckpointIngestInfo.cpp

```diff
@@ -47,12 +46,8 @@ CheckpointIngestInfoPtr CheckpointIngestInfo::restore(
     Page page = uni_ps->read(page_id, nullptr, snapshot, /*throw_on_not_exist*/ false);
     if (!page.isValid())
     {
-        throw Exception(
-            ErrorCodes::LOGICAL_ERROR,
-            "Failed to restore CheckpointIngestInfo, region_id={} peer_id={} store_id={}",
-            region_id,
-            peer_id,
-            tmt.getKVStore()->getStoreID(std::memory_order_relaxed));
+        // The restore failed, we can safely return null here to make FAP fallback.
```

Comment:
Isn't `CheckpointIngestInfo::restore` used to continue the FAP phase 2 after restart? Why it is safe now

Reply:
The problem here is when received a snapshot, we don't know whether it is a fap snapshot or a real empty snapshot, especially after restarted.